galaxy_ng/app/access_control/access_policy.py

```diff
@@ -62,13 +63,34 @@ class CollectionRemoteAccessPolicy(AccessPolicyBase):
 class UserAccessPolicy(AccessPolicyBase):
     NAME = 'UserViewSet'
 
-
-class MyUserAccessPolicy(AccessPolicyBase):
-    NAME = 'MyUserViewSet'
-
     def is_current_user(self, request, view, action):
         return request.user == view.get_object()
 
+    def has_groups_param_obj_perms(self, request, view, action, permission):
+        """
+        Checks if the current user has object-level permission on the ``groups`` object.
+
+        The object in this case is the one specified by the ``remote`` parameter. For example when
+        syncing the ``remote`` parameter is passed in as an argument.
+
+        """
+
+        serializer = UserSerializer(data=request.data, context={"request": request})
+
+        try:
+            serializer.is_valid(raise_exception=True)
+        except Exception as exc:
+            log.exception(exc)
+            raise
```

Comment:
Something about this seems broken. I get a `A user with that username already exists` message when I try to edit users and when I try to add a new group I get this 500

```
api_1               | pulp: django.request:ERROR: Internal Server Error: /api/automation-hub/_ui/v1/users/3/
api_1               | Traceback (most recent call last):
api_1               |   File "/venv/lib64/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
api_1               |     response = get_response(request)
api_1               |   File "/venv/lib64/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
api_1               |     response = self.process_exception_by_middleware(e, request)
api_1               |   File "/venv/lib64/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
api_1               |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
api_1               |   File "/venv/lib64/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
api_1               |     return view_func(*args, **kwargs)
api_1               |   File "/venv/lib64/python3.6/site-packages/rest_framework/viewsets.py", line 114, in view
api_1               |     return self.dispatch(request, *args, **kwargs)
api_1               |   File "/venv/lib64/python3.6/site-packages/rest_framework/views.py", line 505, in dispatch
api_1               |     response = self.handle_exception(exc)
api_1               |   File "/venv/lib64/python3.6/site-packages/rest_framework/views.py", line 465, in handle_exception
api_1               |     self.raise_uncaught_exception(exc)
api_1               |   File "/venv/lib64/python3.6/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
api_1               |     raise exc
api_1               |   File "/venv/lib64/python3.6/site-packages/rest_framework/views.py", line 502, in dispatch
api_1               |     response = handler(request, *args, **kwargs)
api_1               |   File "/venv/lib64/python3.6/site-packages/rest_framework/mixins.py", line 67, in update
api_1               |     serializer.is_valid(raise_exception=True)
api_1               |   File "/venv/lib64/python3.6/site-packages/rest_framework/serializers.py", line 234, in is_valid
api_1               |     self._validated_data = self.run_validation(self.initial_data)
api_1               |   File "/venv/lib64/python3.6/site-packages/rest_framework/serializers.py", line 433, in run_validation
api_1               |     value = self.to_internal_value(data)
api_1               |   File "/src/galaxy_ng/galaxy_ng/app/api/ui/serializers/user.py", line 88, in to_internal_value
api_1               |     for field in group:
api_1               | TypeError: 'int' object is not iterable
```

Reply:
What user is the request being made as?   Did the request end up being a POST or a PATCH?

Reply:
Hmm, I'm seeing variation of the user validation failing now.  I didn't seem to hit that before so I'm not sure what is going on.

ie, something like 

```
 galaxyapi PATCH /api/automation-hub/_ui/v1/users/18/ groups:='[{"id":10, "name":"rh-identity-account:99996"}]'
```

Was working before. 

I suspect something is preventing the "partial" users from being valid now. I don't think I changed how it was called.
Investigating...

Reply:
Yeah, I lost a 'partial=True' for that serializers somewhere. Updated.